core/src/main/java/com/linecorp/armeria/common/logging/RequestContextExporterBuilder.java

```diff
@@ -128,30 +143,54 @@ public boolean containsAttribute(AttributeKey<?> key) {
     @Deprecated
     public Map<String, AttributeKey<?>> getAttributes() {
         return attrs.stream().collect(
-                toImmutableMap(e -> e.exportKey.substring(PREFIX_ATTRS.length()), e -> e.key));
+                toImmutableMap(e -> {
+                    if (e.exportKey.startsWith(PREFIX_ATTRS)) {
+                        return e.exportKey.substring(PREFIX_ATTRS.length());
+                    }
+                    return e.exportKey;
+                }, e -> e.key));
     }
 
     /**
      * Adds the specified HTTP request header name to the export list.
      */
     public RequestContextExporterBuilder addHttpRequestHeader(CharSequence name) {
-        addHttpHeader(PREFIX_HTTP_REQ_HEADERS, httpReqHeaders, requireNonNull(name, "name"));
+        final AsciiString key = toHeaderName(requireNonNull(name, "name"));
+        final String exportKey = PREFIX_HTTP_REQ_HEADERS + key;
+        httpReqHeaders.add(new ExportEntry<>(key, exportKey));
+        return this;
+    }
+
+    /**
+     * Adds the specified HTTP request header name to the export list.
+     * The specified {@code alias} is used for the export key.
+     */
+    public RequestContextExporterBuilder addHttpRequestHeader(CharSequence headerName, String alias) {
+        requireNonNull(headerName, "headerName");
+        requireNonNull(alias, "alias");
+        httpReqHeaders.add(new ExportEntry<>(toHeaderName(headerName), alias));
         return this;
     }
 
     /**
      * Adds the specified HTTP response header name to the export list.
      */
     public RequestContextExporterBuilder addHttpResponseHeader(CharSequence name) {
-        addHttpHeader(PREFIX_HTTP_RES_HEADERS, httpResHeaders, requireNonNull(name, "name"));
+        final AsciiString key = toHeaderName(requireNonNull(name, "name"));
+        final String exportKey = PREFIX_HTTP_RES_HEADERS + key;
+        httpResHeaders.add(new ExportEntry<>(key, exportKey));
         return this;
```

Comment:
Ditto - Could delegate to `addHttpResponseHeader(CharSequence, String)`.

Reply:
It could but it will call `toHeaderName(headerName)` twice.
```java
    public RequestContextExporterBuilder addHttpResponseHeader(CharSequence headerName) {
        final AsciiString key = toHeaderName(requireNonNull(headerName, "headerName"));
        final String exportKey = PREFIX_HTTP_RES_HEADERS + key;
        return addHttpResponseHeader(headerName, exportKey);
    }
```

Reply:
Will add another private method. :-)